src/test/java/uk/ac/ebi/eva/evaseqcol/service/SeqColExtendedDataServiceTest.java

```diff
@@ -346,12 +346,32 @@ public SeqColExtendedDataEntity constructSeqColSequencesObject(AssemblySequenceE
     }
 
     /**
-     * Return the 3 seqcol objects (names, lengths and sequences) of the given naming convention*/
-    List<SeqColExtendedDataEntity> constructLevelTwoSeqCols(AssemblyEntity assemblyEntity, AssemblySequenceEntity sequenceEntity,
-                                                            SeqColEntity.NamingConvention convention) throws IOException {
+     * Return the 3 extended objects (names, lengths and sequences) of the given naming convention*/
+    List<SeqColExtendedDataEntity> constructExtendedSeqColDataList(AssemblyEntity assemblyEntity, AssemblySequenceEntity assemblySequenceEntity,
+                                                            SeqColEntity.NamingConvention convention, String accession) throws IOException {
         // Sorting the chromosomes' list (assemblyEntity) and the sequences' list (sequencesEntity) in the same order
-        sortReportAndSequencesBySequenceIdentifier(assemblyEntity, sequenceEntity, GCA_ACCESSION);
-        List<SeqColExtendedDataEntity> entities = new Li
+        sortReportAndSequencesBySequenceIdentifier(assemblyEntity, assemblySequenceEntity, accession);
+        return Arrays.asList(
+                constructSeqColSequencesObject(assemblySequenceEntity),
+                constructSeqColNamesObject(assemblyEntity, convention),
+                constructSeqColLengthsObject(assemblyEntity)
+        );
+    }
+
+    /**
+     * Construct and return a Level Two (with exploded data) SeqCol entity out of the given assemblyEntity and the
+     * assemblySequencesEntity*/
+    SeqColLevelTwoEntity constructSeqColLevelTwo(AssemblyEntity assemblyEntity, AssemblySequenceEntity assemblySequenceEntity,
+                                                 SeqColEntity.NamingConvention convention, String accession) throws IOException {
+        SeqColLevelTwoEntity seqColLevelTwo = new SeqColLevelTwoEntity();
+        sortReportAndSequencesBySequenceIdentifier(assemblyEntity, assemblySequenceEntity, accession);
+        SeqColExtendedDataEntity extendedNamesData = constructSeqColNamesObject(assemblyEntity, convention);
+        SeqColExtendedDataEntity extendedLengthsData = constructSeqColLengthsObject(assemblyEntity);
+        SeqColExtendedDataEntity extendedSequencesData = constructSeqColSequencesObject(assemblySequenceEntity);
+        seqColLevelTwo.setNames(extendedNamesData.getObject().getObject());
+        seqColLevelTwo.setLengths(extendedLengthsData.getObject().getObject());
+        seqColLevelTwo.setSequences(extendedSequencesData.getObject().getObject());
+        return seqColLevelTwo;
```

Comment:
I am confused as to why we are re-writing the function already defined in SeqColExtendedDataService?